### PR TITLE
[JSPI] - Assert that JSPI is supported by the environment.

### PR DESCRIPTION
**Pull request by brendandahl**

### CODE REVIEW

File: src/library_async.js [hunk 1]
```diff
@@ -40,6 +40,9 @@ addToLibrary({
     instrumentWasmImports(imports) {
 #if ASYNCIFY_DEBUG
       dbg('asyncify instrumenting imports');
+#endif
+#if ASSERTIONS && ASYNCIFY == 2
+      assert('Suspender' in WebAssembly, 'JSPI must be enabled.');
```
Comment (sbc100) [inline src/library_async.js:45]: How about "JSPI not support by current browser.  Perhaps it needs to be enabled via flags?"

Comment (brendandahl) [reply src/library_async.js:45]: /s/browser/runtime? This may be node too. What do we usually refer to the runtime as?

Comment (brendandahl) [reply src/library_async.js:45]: I guess it's environment in settings.

Comment (sbc100) [reply src/library_async.js:45]: Fair enough.

File: src/library_async.js [hunk 2]
```diff
@@ -40,6 +40,9 @@ addToLibrary({
     instrumentWasmImports(imports) {
 #if ASYNCIFY_DEBUG
       dbg('asyncify instrumenting imports');
+#endif
+#if ASSERTIONS && ASYNCIFY == 2
+      assert('Suspender' in WebAssembly, 'JSPI not support by current environment. Perhaps it needs to be enabled via flags?');
```
Comment (sbc100) [inline src/library_async.js:45]: /support/supported

**Review by sbc100 — APPROVED** [review]

**Pull request merged**